screens/Challenges/ToDoScreen.js

```diff
@@ -1,48 +1,119 @@
 import React, { useState, useEffect } from "react";
 import { View, StyleSheet, Image } from "react-native";
 import { ScrollView } from "react-native";
+import moment from "moment";
 
 import KitText from "../../components/KitText";
-import ChallengesScreen from "../ChallengesScreen";
+import OpenChallengeCard from "../../components/OpenChallengeCard";
 import ChallengeTodo from "../../components/challenges/ChallengeTodo";
 
 import Colors from "../../constants/Colors";
-import Fonts from "../../constants/Fonts";
 
-import { getAllAssignedChallenges } from "../../utils/db/challenges";
+import {
+  getAllAssignedChallenges,
+  openChallenge
+} from "../../utils/db/challenges";
 import { getUserId } from "../../utils/auth/auth";
 
 export default function ToDoScreen() {
-  const [challenges, setChallenges] = useState([]);
+  const [openedChallenges, setOpenedChallenges] = useState([]);
+  const [unopenedChallenges, setUnopenedChallenges] = useState([]);
+  const [challengeCards, setChallengeCards] = useState([]);
 
   function renderTodos() {
-    let todos = []
+    let todos = [];
     let alternator = true;
-    for (challenge of challenges) {
+    for (challenge of openedChallenges) {
       // console.log(challenge)
       todos.push(
-        <ChallengeTodo key={challenge.teamId} challenge={challenge} mainColor={(alternator ? Colors.KIT_LIGHT_ORANGE : Colors.KIT_GREEN)} onPress={() => {alert("Hello")}}/>
-      )
+        <ChallengeTodo
+          key={challenge.teamId}
+          challenge={challenge}
+          mainColor={alternator ? Colors.KIT_LIGHT_ORANGE : Colors.KIT_GREEN}
+          onPress={() => {
+            alert("Hello");
+          }}
+        />
+      );
       alternator = !alternator;
     }
-    return todos
+    return todos;
   }
 
   useEffect(() => {
-    currentUser = getUserId();
+    const currentUser = getUserId();
     getAllAssignedChallenges(currentUser).then(challenges => {
-      setChallenges(challenges)
-    })
+      const openedChallenges = [];
+      const unopenedChallenges = [];
+      // Filter opened and unopened challenges
+      for (const challenge of challenges) {
+        let opened = false;
+        if (challenge.hasOwnProperty("opened")) {
+          const openedUserObjects = challenge.opened;
+          for (const openedUser of Object.keys(openedUserObjects)) {
+            if (openedUser === currentUser) {
+              opened = true;
+              break;
+            }
+          }
+        }
+        if (opened) {
+          openedChallenges.push(challenge);
+        } else {
+          unopenedChallenges.push(challenge);
+        }
+      }
+      setOpenedChallenges(openedChallenges);
+      setUnopenedChallenges(unopenedChallenges);
+      setChallengeCards(unopenedChallenges);
+    });
   }, []);
 
+  const getTimeLeftInMilliseconds = assignedTime => {
+    // 24 hours in seconds
+    const TIME_TO_OPEN = 86400000;
```

Comment:
All users will have 24 hours to "reveal" a challenge?

May be nice to include a comment about the current design seeing how you are declaring a hardcoded 
constant